### PR TITLE
Assignement 1

### PR DESCRIPTION
**Pull request by zeevso**
Ready to submit

### CODE REVIEW

File: exercise2.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        return self.min_age <= age < self.max_age
+
+
+def read_json():
+    with open(INPUT_FILE) as file:
+        return json.load(file)
+
+
+def init_ranges():
+    data = read_json()
```
Comment (li0nbelenky) [inline exercise2.py]: why do you run read_json() again from here,
If you already run it during init and pass it to a different function but not this one?
Just trying to understand, maybe you have a good reason.

Comment (zeevso) [reply exercise2.py]: No, just a bug. That is the reason I transfered data as parameter before and was very surprised as you pointed that it is redundant.  Do you expect for the fixed version?

Thanx

Comment (li0nbelenky) [reply exercise2.py]: your submission is very a good, so just this small touch to make it perfect :)

**Comment by li0nbelenky** [comment]
@eyalstoler i approve this submission.

File: exercise2.py (hunk 1 above)
Comment (li0nbelenky) [reply exercise2.py]: @zeevso One last thing,
There are many places in your code where you reference keys in dict,
places in array, create http request or read a file and did not protect it with try/except.
Note that catching exceptions in python needs to be precise.
For example:
new_list = [1,2,3,4]
try:
    print(new_list[66])
except IndexError:
    print(f'Bad index in list {new_list}')

**Pull request closed without merging**